### ISSUE TITLE
CRITICAL: Investigate reranker regression (73.3% → 56.7% accuracy drop)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#701]
## Problem
Neural reranker is **reducing accuracy** instead of improving it.

## Evidence
Test results on 30-question diverse set:

| Configuration | Accuracy | WHO Accuracy |
|---------------|----------|--------------|
| **Without reranker** | 73.3% (22/30) | 82.4% (14/17) |
| **With reranker** | 56.7% (17/30) | 52.9% (9/17) |
| **Regression** | **-16.6 points** | **-29.5 points** |

## Files
- Test results: `/tmp/and_query_full_test.txt` (no reranker, 73.3%)
- Test results: `/tmp/full_pipeline_test.txt` (with reranker, 56.7%)
- Reranker model: `models/reranker/best_model.pt` (180K params)
- Reranker code: `klareco/models/reranker.py`

### 1. Check Training Data Quality
- Was reranker trained on correct positive/negative pairs?
- Is there label noise or swapped labels?
- Check training loss curves

### 2. Check Model Loading
- Are embeddings loaded correctly?
- Are frozen embeddings actually frozen?
- Dimension mismatch? (64D vs 128D)

### 3. Check Scoring Logic
- Is reranker inverting scores (high → low)?
- Are scores normalized correctly?
- Debug on single question: compare BM25 rank vs reranker rank

### 4. Test Reranker Independently
```bash
# Score query-doc pairs directly
python scripts/test_reranker.py "Kiu estis Roosevelt?"
# Should rank Roosevelt sentences higher than random sentences
```

### 5. Ablation Test
- Try reranker with different thresholds
- Try top-k cutoff (only rerank top 10, keep rest)
- Compare reranker scores to BM25 scores

## Quick Debug
```python
# In klareco/rag/extractive_answering.py line 283
logger.info(f"Reranker scores: {scores[:5]}")  # Top 5 scores
logger.info(f"BM25 scores: {[s['score'] for s in sentences[:5]]}")
```

## Expected Root Cause (Hypotheses)
1. **Label swap**: Positive/negative pairs swapped during training
2. **Score inversion**: Model learned to give low scores to relevant docs
3. **Dimension mismatch**: Loading 64D embeddings into 128D model
4. **Overfitting**: Model memorized training noise

## Priority
**P0 - Critical**: This is blocking deployment of reranker. Current best practice is `--no-rerank`.

## Workaround
Use `--no-rerank` flag until issue is resolved:
```bash
python scripts/evaluate_extractive_qa.py --no-rerank
```

## Related
- Reranker was working in previous versions
- This regression appeared after speed optimizations (correlation, not causation)
- M1 also has accuracy issues (#687)
